### PR TITLE
Active Directory Query v2 - Update Required Permissions

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-26104)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 1]
```diff
@@ -355,6 +386,10 @@ Creates an Active Directory contact.
 #### Base Command
 
 `ad-create-contact`
+
+##### Required Permissions
+Requires `Create, delete, and manage user accounts` permissions.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:391]: R u sure there is no way permission for only for contacts (and not for user accounts)?
image

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 2]
```diff
@@ -390,6 +425,10 @@ Updates attributes of an existing Active Directory contact.
 #### Base Command
 
 `ad-update-contact`
+
+##### Required Permissions
+Requires `Create, delete, and manage user accounts` permissions.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:430]: same here

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 3]
```diff
@@ -578,6 +629,10 @@ Modifies the user organizational unit within a domain.
 #### Base Command
 
 `ad-modify-user-ou`
+
+##### Required Permissions
+Requires `Read userAccountControl` and `write userAccountControl` permissions.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:634]: Please check this - might be even better
https://learn.microsoft.com/en-us/answers/questions/973272/delegate-help-desk-users-permission-to-move-users

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 4]
```diff
@@ -549,6 +596,10 @@ Modifies the computer organizational unit within a domain.
 #### Base Command
 
 `ad-modify-computer-ou`
+
+##### Required Permissions
+Requires `Full Control` permissions from `Computer objects`.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:601]: same here 
https://learn.microsoft.com/en-us/answers/questions/973272/delegate-help-desk-users-permission-to-move-users

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 5]
```diff
@@ -1016,6 +1107,10 @@ Used in the IAM premium pack.
 #### Base Command
 
 `iam-disable-user`
+
+##### Required Permissions
+Requires `Create, delete, and manage user accounts` permissions.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:1112]: should be changed to this 
`Requires Read userAccountControl and write userAccountControl permissions.`
Similar to `ad-disable-account`

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 6]
```diff
@@ -295,6 +319,10 @@ Updates attributes of an existing Active Directory user.
 #### Base Command
 
 `ad-update-user`
+
+##### Required Permissions
+Requires `Create, delete, and manage user accounts` permissions.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:324]: Is there an option to use Read/Write All user properties?

**Review by ssokolovich — COMMENTED** [review]
Looks fine - pls check my comments.

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md (hunk 1 above)
Comment (adi88d) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:391]: I dint find anything about this specific permission but `Create, delete, and manage user accounts` should be enough

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md (hunk 2 above)
Comment (adi88d) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:430]: I didnt find anything about this specific permission but `Create, delete, and manage user accounts` should be enough

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md (hunk 6 above)
Comment (adi88d) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:324]: I didnt find anything about this specific permission but `Create, delete, and manage user accounts` should be enough

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md (hunk 1 above)
Comment (ssokolovich) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:391]: I don't like the option for delete here - should be only READ / Write all properties.
And only for contact objects.

image
image

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md (hunk 6 above)
Comment (ssokolovich) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:324]: Same as above, only Read / Write all properties ( without delete) 
to `user objects`

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 7]
```diff
@@ -355,6 +386,10 @@ Creates an Active Directory contact.
 #### Base Command
 
 `ad-create-contact`
+
+##### Required Permissions
+Requires `Write All Properties` and `Read All Properties` permission from `Contact objects`.
```
Comment (ssokolovich) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:391]:
```suggestion
Requires `full control` permission from `Contact objects`.
```

**Review by ssokolovich — APPROVED** [review]

**Pull request merged**